### DISCUSSION TITLE
Support for private/personal scripts?

### DISCUSSION THREAD

**Discussion by inklesspen** [AprilSylph/XKit-Rewritten#1085] [Suggestion Box]
There are some tweaks I would make to the Tumblr UI, which would really only be useful for my personal usage habits, so I don't think they belong in the XKit codebase.

However, it would be nice to be able to take advantage of XKit's API, such as meatball menu utilities, css map helpers, and so on. If I implement my script as a userscript, I will have to reimplement (or copy) this functionality. And if I add my script to my local XKit repo, I would have to keep my repo up to date with the main XKit repo by hand.

Perhaps it would be feasible to support adding a user-defined script to the extension-as-distributed?

**Comment by AprilSylph** [comment]
Would be cool, but the way XKit 7 implements this is by running everything via evaluation of strings in the privileged extension context, which Manifest V3 is very against allowing. I'm not familiar enough with the new userscripts API to dream of a different way forward.

**Reply by nightpool** [reply]
My understanding is that https://developer.chrome.com/docs/extensions/reference/scripting/ and https://github.com/w3c/webextensions/blob/main/proposals/user-scripts-api.md are the current state of the art for what is allowed for MV3, with the former being "what is currently allowed" and the latter being "what will be allowed eventually" prior to MV2's removal.

I think both of these proposals would allow for an XKit-7-like user script manager but obviously it would be a lot of work to implement.

**Reply by RFLEpicGamer** [reply]
And talking about Manifest v3, do the creator plan to switch this extension to manifest v3?

**Reply by marcustyphoon** [reply]
My guess is that we would if forced to and would not if not forced to (the Chrome forced-transition date keeps getting pushed back). This isn't like an adblocker where MV3 would significantly change how the code works/what it does, so it's pretty whatever.
